chainer/training/updaters/parallel_updater.py

```diff
@@ -120,20 +121,23 @@ def update_core(self):
             loss_func = self.loss_func or model
 
             with function.force_backprop_mode():
-                if isinstance(in_arrays, tuple):
-                    loss = loss_func(*in_arrays)
-                elif isinstance(in_arrays, dict):
-                    loss = loss_func(**in_arrays)
-                else:
-                    loss = loss_func(in_arrays)
+                with cuda.get_device_from_id(self._devices[model_key]):
+                    if isinstance(in_arrays, tuple):
+                        loss = loss_func(*in_arrays)
+                    elif isinstance(in_arrays, dict):
+                        loss = loss_func(**in_arrays)
+                    else:
+                        loss = loss_func(in_arrays)
+
             losses.append(loss)
 
         # For _uninitialized_params
         for model in six.itervalues(self._models):
             model.cleargrads()
 
         for loss in losses:
-            loss.backward(loss_scale=self.loss_scale)
+            with cuda.get_device_from_array(loss.array):
```

Comment:
I noticed this is not needed as device is now selected properly in chainer.Variable.backward. (#4725)